src/server/scripts/Northrend/Ulduar/Ulduar/boss_yogg_saron.cpp

```diff
@@ -1419,7 +1430,9 @@ class npc_descend_into_madness : public CreatureScript
             {
                 if (!result)
                     return;
+
                 clicker->RemoveAurasDueToSpell(SPELL_BRAIN_LINK);
+                DoCast(clicker, IllusionSpells[_instance->GetData(DATA_ILLUSION)], true);
```

Comment:
afaik only if you use SetData do change it, other places where this GetData is used does not perform any check too, and even if return a weird value(which should not happen), DoCast do check already checks if the spell passed is valid.